partitura/io/musescore.py

```diff
@@ -36,38 +36,69 @@ class FileImportException(Exception):
     pass
 
 
-def find_musescore3():
-    # # possible way to detect MuseScore... executable
-    # for p in os.environ['PATH'].split(':'):
-    #     c = glob.glob(os.path.join(p, 'MuseScore*'))
-    #     if c:
-    #         print(c)
-    #         break
-
-    result = shutil.which("musescore")
-
-    if result is None:
-        result = shutil.which("musescore3")
-
-    if result is None:
-        result = shutil.which("mscore")
-
+def find_musescore_version(version=4):
+    """Find the path to the MuseScore executable for a specific version.
+    If version is a empty string it tries to find an unspecified version of
+    MuseScore which is used in some systems.
+    """
+    result = shutil.which(f"musescore{version}")
     if result is None:
-        result = shutil.which("mscore3")
-
+        result = shutil.which(f"mscore{version}")
     if result is None:
         if platform.system() == "Linux":
             pass
-
         elif platform.system() == "Darwin":
-            result = shutil.which("/Applications/MuseScore 3.app/Contents/MacOS/mscore")
-
+            result = shutil.which(
+                f"/Applications/MuseScore {version}.app/Contents/MacOS/mscore"
+            )
         elif platform.system() == "Windows":
-            result = shutil.which(r"C:\Program Files\MuseScore 3\bin\MuseScore3.exe")
+            result = shutil.which(
+                rf"C:\Program Files\MuseScore {version}\bin\MuseScore{version}.exe"
+            )
 
     return result
 
 
+def find_musescore():
+    """Find the path to the MuseScore executable.
+
+    This function first tries to find the executable for MuseScore 4,
+    then for MuseScore 3, and finally for any version of MuseScore.
+
+    Returns
+    -------
+    str
+        Path to the MuseScore executable
+
+    Raises
+    ------
+    MuseScoreNotFoundException
+        When no MuseScore executable was found
+    """
+
+    mscore_exec = find_musescore_version(version=4)
+    if not mscore_exec:
+        mscore_exec = find_musescore_version(version=3)
+        if mscore_exec:
+            warnings.warn(
+                "Only Musescore 3 is installed. Consider upgrading to musescore 4."
+            )
+        else:
+            mscore_exec = find_musescore_version(version="")
+            if mscore_exec:
+                warnings.warn(
+                    "A unspecified version of MuseScore was found. Consider upgrading to musescore 4."
+                )
+            else:
+                raise MuseScoreNotFoundException()
+    if "DISPLAY" not in os.environ:
```

Comment:
I end up at this exception on my laptop. (it has a display)

Reply:
After internal discussion, the implemented solution is to check if the system is Linux and then check for display in `os.environ`. This is useful for most servers running Linux. The workaround for Windows or MacOS would be to use tkinter (a built-in library) but since such servers are not so common we will not implement it for now.